libs/developers-tx-result-adapter-spring-support/build.gradle.kts

```diff
@@ -69,83 +69,102 @@ repositories {
     mavenCentral()
 }
 
-val sourcesJar by tasks.creating(Jar::class) {
-    archiveClassifier.set("sources")
-    from(sourceSets.main.map { it.allSource })
-}
+fun setPublishingContext() {
+    fun checkEnvironmentAndThrow() {
+        requiredEnvironments.forEach {
+            System.getenv(it) ?: throw GradleException("Environment need to be set : $it")
+        }
+    }
+    checkEnvironmentAndThrow()
 
-val javadocJar by tasks.creating(Jar::class) {
-    val javadoc = tasks.named("javadoc")
-    dependsOn(javadoc)
-    archiveClassifier.set("javadoc")
-    from(javadoc)
-}
+    val sourcesJar by tasks.creating(Jar::class) {
+        archiveClassifier.set("sources")
+        from(sourceSets.main.map { it.allSource })
+    }
 
-publishing {
-    repositories {
-        maven {
-            name = "ossRelease"
-            url = uri(releasesRepoUrl)
-            credentials {
-                username = System.getenv(MAVEN_USERNAME)
-                password = System.getenv(MAVEN_PASSWORD)
+    val javadocJar by tasks.creating(Jar::class) {
+        val javadoc = tasks.named("javadoc")
+        dependsOn(javadoc)
+        archiveClassifier.set("javadoc")
+        from(javadoc)
+    }
+
+    publishing {
+        repositories {
+            maven {
+                if (System.getenv(DEPLOY_VERSION).endsWith("-SNAPSHOT")) {
+                    name = snapshotRepoName
+                    url = uri(snapshotRepoUrl)
+                } else {
+                    name = releaseRepoName
+                    url = uri(releasesRepoUrl)
+                }
+
+                credentials {
+                    username = System.getenv(SONATYPE_USERNAME)
+                    password = System.getenv(SONATYPE_PASSWORD)
+                }
             }
         }
-    }
 
-    publications {
-        create<MavenPublication>(project.name) {
-            from(components["kotlin"])
+        publications {
+            create<MavenPublication>(project.name) {
+                from(components["kotlin"])
 
-            artifact(sourcesJar)
-            artifact(javadocJar)
+                artifact(sourcesJar)
+                artifact(javadocJar)
 
-            groupId = publishGroupId
-            artifactId = "developers-tx-result-adapter"
-            version = System.getenv("DEPLOY_VERSION")
+                groupId = publishGroupId
+                artifactId = "tx-result-adapter-spring-support"
+                version = System.getenv("DEPLOY_VERSION")
 
-            pom {
-                packaging = "jar"
-                name.set("pom: nexus-deploy-test-name")
-                description.set("pom: nexus-deploy-test-desc")
-                url.set(gitRepositoryUrl)
+                pom {
+                    packaging = "jar"
+                    name.set("pom: nexus-deploy-test-name")
```

Comment:
I guess we also need to update this to `line-blockchain-developers-tx-result-adapter-spring-support`, don't we?

Reply:
Thank you. I'll fix it.